### ISSUE TITLE
Upgrading DB from 12 to 13 fails with WARNING Table 'res_lang': unable to set NOT NULL on column 'url_code'!

### ISSUE THREAD

**Issue by jinnko** [odoo/odoo#39381]
### Impacted versions: 13.0-20191022

### Steps to reproduce:

1. Stop Odoo-12 docker container
2. Pull Odoo-13 docker container
3. Run DB upgrade command via container, observe error messages

### Current behavior:

```
# odoo -u all -d odoo -r odoo -w ... --db_host=db --stop-after-init                                                                                                                                                                                                                                                                                
2019-10-25 11:47:24,610 419 INFO ? odoo: Odoo version 13.0-20191022                                                                                                                                                                                                                                                                                                                    
2019-10-25 11:47:24,610 419 INFO ? odoo: Using configuration file at /etc/odoo/odoo.conf                                                                                                                                                                                                                                                                                                
2019-10-25 11:47:24,610 419 INFO ? odoo: addons paths: ['/usr/lib/python3/dist-packages/odoo/addons', '/var/lib/odoo/addons/13.0', '/mnt/extra-addons']                                                                                                                                                                                                                                 
2019-10-25 11:47:24,610 419 INFO ? odoo: database: odoo@db:default                                                                                                                                                                                                                                                                                                                     
2019-10-25 11:47:24,708 419 INFO ? odoo.addons.base.models.ir_actions_report: Will use the Wkhtmltopdf binary at /usr/local/bin/wkhtmltopdf                                                                                                                                                                                                                                            
2019-10-25 11:47:24,779 419 INFO odoo odoo.modules.loading: loading 1 modules...                                                                                                                                                                                                                                                                                                        
2019-10-25 11:47:24,849 419 INFO odoo odoo.modules.registry: module base: creating or updating database tables                                                                                                                                                                                                                                                                         
2019-10-25 11:47:25,253 419 ERROR odoo odoo.sql_db: bad query: ALTER TABLE "res_lang" ALTER COLUMN "url_code" SET NOT NULL                                                                                                                                                                                                                                                             
ERROR: column "url_code" contains null values                                                                                                                                                                                                                                                                                                                                           
                                                                                                                                                                                                                                                                                                                                                                                       
2019-10-25 11:47:25,253 419 WARNING odoo odoo.schema: Table 'res_lang': unable to set NOT NULL on column 'url_code'!                                                                                                                                                                                                                                                                   
If you want to have it, you should update the records and execute manually:                                                                                                                                                                                                                                                                                                             
ALTER TABLE "res_lang" ALTER COLUMN "url_code" SET NOT NULL                                                                                                                                                                                                                                                                                                                             
Traceback (most recent call last):                                                                                                                                                                                                                                                                                                                                                     
  File "/usr/lib/python3/dist-packages/odoo/tools/sql.py", line 112, in set_not_null                                                                                                                                                                                                                                                                                                    
    cr.execute(query)                                                                                                                                                                                                                                                                                                                                                                   
  File "/usr/lib/python3/dist-packages/odoo/sql_db.py", line 163, in wrapper                                                                                                                                                                                                                                                                                                            
    return f(self, *args, **kwargs)                                                                                                                                                                                                                                                                                                                                                    
  File "/usr/lib/python3/dist-packages/odoo/sql_db.py", line 240, in execute                                                                                                                                                                                                                                                                                                           
    res = self._obj.execute(query, params)                                                                                                                                                                                                                                                                                                                                             
psycopg2.errors.NotNullViolation: column "url_code" contains null values                                                                                                                                                                                                                                                                                                                
                                                                                                                                                                                                                                                                                                                                                                                       
2019-10-25 11:47:26,202 419 INFO odoo odoo.modules.loading: loading base/data/res.lang.csv                                                                                                                                                                                                                                                                                              
2019-10-25 11:47:26,251 419 ERROR odoo odoo.models: Error while loading record                                                                                                                                                                                                                                                                                                         
Traceback (most recent call last):                                                                                                                                                                                                                                                                                                                                                     
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 921, in flush                                                                                                                                                                                                                                                                                                             
    rec = self._load_records([rec_data], mode == 'update')                                                                                                                                                                                                                                                                                                                              
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3963, in _load_records                                                                                                                                                                                                                                                                                                    
    data['record']._load_records_write(data['values'])                                                                                                                                                                                                                                                                                                                                  
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3901, in _load_records_write                                                                                                                                                                                                                                                                                              
    self.write(values)                                                                                                                                                                                                                                                                                                                                                                  
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/res_lang.py", line 235, in write                                                                                                                                                                                                                                                                                         
    raise UserError(_("Language code cannot be modified."))                                                                                                                                                                                                                                                                                                                            
odoo.exceptions.UserError: ('Language code cannot be modified.', '')                                                                                                                                                                                                                                                                                                                    
2019-10-25 11:47:26,294 419 ERROR odoo odoo.models: Error while loading record                                                                                                                                                                                                                                                                                                         
Traceback (most recent call last):                                                                                                                                                                                                                                                                                                                                                      
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 921, in flush                                                                                                                                                                                                                                                                                                             
    rec = self._load_records([rec_data], mode == 'update')                                                                                                                                                                                                                                                                                                                             
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3963, in _load_records                                                                                                                                                                                                                                                                                                     
    data['record']._load_records_write(data['values'])                                                                                                                                                                                                                                                                                                                                  
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3901, in _load_records_write                                                                                                                                                                                                                                                                                               
    self.write(values)                                                                                                                                                                                                                                                                                                                                                                 
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/res_lang.py", line 235, in write                                                                                                                                                                                                                                                                                         
    raise UserError(_("Language code cannot be modified."))                                                                                                                                                                                                                                                                                                                            
odoo.exceptions.UserError: ('Language code cannot be modified.', '')                                                                                                                                                                                                                                                                                                                    
2019-10-25 11:47:26,441 419 WARNING odoo odoo.modules.loading: Transient module states were reset                                                                                                                                                                                                                                                                                       
2019-10-25 11:47:26,441 419 ERROR odoo odoo.modules.registry: Failed to load registry                                                                                                                                                                                                                                                                                                  
Traceback (most recent call last):                                                                                                                                                                                                                                                                                                                                                     
  File "/usr/lib/python3/dist-packages/odoo/modules/registry.py", line 86, in new                                                                                                                                                                                                                                                                                                      
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)                                                                                                                                                                                                                                                                                                         
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 361, in load_modules                                                                                                                                                                                                                                                                                              
    report=report, models_to_check=models_to_check)                                                                                                                                                                                                                                                                                                                                    
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 222, in load_module_graph                                                                                                                                                                                                                                                                                        
    load_data(cr, idref, mode, kind='data', package=package, report=report)                                                                                                                                                                                                                                                                                                            
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 68, in load_data                                                                                                                                                                                                                                                                                                  
    tools.convert_file(cr, package.name, filename, idref, mode, noupdate, kind, report)                                                                                                                                                                                                                                                                                                 
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 733, in convert_file                                                                                                                                                                                                                                                                                                
    convert_csv_import(cr, module, pathname, fp.read(), idref, mode, noupdate)                                                                                                                                                                                                                                                                                                         
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 780, in convert_csv_import                                                                                                                                                                                                                                                                                         
    raise Exception(_('Module loading %s failed: file %s could not be processed:\n %s') % (module, fname, warning_msg))                                                                                                                                                                                                                                                                
Exception: Module loading base failed: file base/data/res.lang.csv could not be processed:                                                                                                                                                                                                                                                                                             
 Unknown error during import: <class 'odoo.exceptions.UserError'>: ('Language code cannot be modified.', '')                                                                                                                                                                                                                                                                           
Unknown error during import: <class 'odoo.exceptions.UserError'>: ('Language code cannot be modified.', '')                                                                                                                                                                                                                                                                            
2019-10-25 11:47:26,442 419 CRITICAL odoo odoo.service.server: Failed to initialize database `odoo`.                                                                                                                                                                                                                                                                                   
Traceback (most recent call last):                                                                                                                                                                                                                                                                                                                                                      
  File "/usr/lib/python3/dist-packages/odoo/service/server.py", line 1146, in preload_registries                                                                                                                                                                                                                                                                                        
    registry = Registry.new(dbname, update_module=update_module)                                                                                                                                                                                                                                                                                                                       
  File "/usr/lib/python3/dist-packages/odoo/modules/registry.py", line 86, in new                                                                                                                                                                                                                                                                                                      
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)                                                                                                                                                                                                                                                                                                          
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 361, in load_modules                                                                                                                                                                                                                                                                                              
    report=report, models_to_check=models_to_check)                                                                                                                                                                                                                                                                                                                                    
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 222, in load_module_graph                                                                                                                                                                                                                                                                                        
    load_data(cr, idref, mode, kind='data', package=package, report=report)                                                                                                                                                                                                                                                                                                             
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 68, in load_data                                                                                                                                                                                                                                                                                                 
    tools.convert_file(cr, package.name, filename, idref, mode, noupdate, kind, report)                                                                                                                                                                                                                                                                                                
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 733, in convert_file                                                                                                                                                                                                                                                                                               
    convert_csv_import(cr, module, pathname, fp.read(), idref, mode, noupdate)                                                                                                                                                                                                                                                                                                          
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 780, in convert_csv_import                        
    raise Exception(_('Module loading %s failed: file %s could not be processed:\n %s') % (module, fname, warning_msg))          
Exception: Module loading base failed: file base/data/res.lang.csv could not be processed:                                                                                                                                                                                                                                                                                              
 Unknown error during import: <class 'odoo.exceptions.UserError'>: ('Language code cannot be modified.', '')                     
Unknown error during import: <class 'odoo.exceptions.UserError'>: ('Language code cannot be modified.', '')                                                                                                                                                                                                                                                                             
2019-10-25 11:47:26,442 419 INFO odoo odoo.service.server: Initiating shutdown                                                                                                                                                                                                                                                                                                          
2019-10-25 11:47:26,442 419 INFO odoo odoo.service.server: Hit CTRL-C again or send a second signal to force the shutdown.     
```

With debug logging enabled:

```
# odoo -u all -d odoo -r odoo -w ... --db_host=db --stop-after-init  --log-level=debug_sql
...
2019-10-25 12:30:11,768 535 DEBUG odoo odoo.sql_db: query: SAVEPOINT "304b9a34f72311e9b82b0242ac110005"                                                                                                                                                                                                                                                                                
2019-10-25 12:30:11,768 535 DEBUG odoo odoo.sql_db: query: ALTER TABLE "res_lang" ALTER COLUMN "url_code" SET NOT NULL                                                                                                                                                                                                                                                                 
2019-10-25 12:30:11,769 535 ERROR odoo odoo.sql_db: bad query: ALTER TABLE "res_lang" ALTER COLUMN "url_code" SET NOT NULL                                                                                                                                                                                                                                                             
ERROR: column "url_code" contains null values                                                                                                                                                                                                                                                                                                                                          
                                                                                                                                                                                                                                                                                                                                                                                       
2019-10-25 12:30:11,769 535 DEBUG odoo odoo.sql_db: query: ROLLBACK TO SAVEPOINT "304b9a34f72311e9b82b0242ac110005"                                                                                                                                                                                                                                                                    
2019-10-25 12:30:11,769 535 WARNING odoo odoo.schema: Table 'res_lang': unable to set NOT NULL on column 'url_code'!                                                                                                                                                                                                                                                                   
If you want to have it, you should update the records and execute manually:                                                                                                                                                                                                                                                                                                            
ALTER TABLE "res_lang" ALTER COLUMN "url_code" SET NOT NULL                                                                                                                                                                                                                                                                                                                            
Traceback (most recent call last):                                                                                                                                                                                                                                                                                                                                                     
  File "/usr/lib/python3/dist-packages/odoo/tools/sql.py", line 112, in set_not_null                                                                                                                                                                                                                                                                                                   
    cr.execute(query)                                                                                                                                                                                                                                                                                                                                                                  
  File "/usr/lib/python3/dist-packages/odoo/sql_db.py", line 163, in wrapper                                                                                                                                                                                                                                                                                                           
    return f(self, *args, **kwargs)                                                                                                                                                                                                                                                                                                                                                    
  File "/usr/lib/python3/dist-packages/odoo/sql_db.py", line 240, in execute                                                                                                                                                                                                                                                                                                           
    res = self._obj.execute(query, params)                                                                                                                                                                                                                                                                                                                                             
psycopg2.errors.NotNullViolation: column "url_code" contains null values                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   ...
```

### Expected behavior:

Clean DB upgrade

### Video/Screenshot link (optional):

**Comment by pedrobaeza** [comment]
Migration between major versions is not included, so that's why you receive that error (or any other).

**Comment by raviprasadmr** [comment]
@jinnko Did you solve this issue? Even I am facing this issue now. Could you give pointers?

**Comment by jinnko** [comment]
@raviprasadmr - After raising this ticket I learned Odoo don't support major version upgrades in the community edition as it's a very complicated effort.  I did however find [OpenUpgrade](https://doc.therp.nl/openupgrade/) which is a community project aimed at handling this.  Unfortunately it doesn't yet fully support the upgrade from 12 to 13, and I haven't had an opportunity to help improve OpenUpgrade yet.  See the OpenUpgrade [Migration scripts for v13](https://github.com/OCA/OpenUpgrade/issues/1733) issue for the latest progress.
